web/core/src/jsMain/kotlin/androidx/compose/web/css/CSSUnits.kt

```diff
@@ -1,38 +1,30 @@
+@file:Suppress("Unused", "NOTHING_TO_INLINE", "NESTED_CLASS_IN_EXTERNAL_INTERFACE", "INLINE_EXTERNAL_DECLARATION", "WRONG_BODY_OF_EXTERNAL_DECLARATION", "NESTED_EXTERNAL_DECLARATION")
+
 package org.jetbrains.compose.web.css
 
-interface CSSSizeValue<out T : CSSUnit> : StylePropertyValue {
+external interface CSSSizeValue<out T : CSSUnit> : StylePropertyValue {
     val value: Float
     val unit: T
-    fun asString(): String = "${value}${unit.value}"
-    fun newUnit(value: Float): CSSSizeValue<T>
 }
 
-private data class CSSUnitValueTyped<out T : CSSUnit>(
+@Suppress("EqualsOrHashCode")
+data class CSSUnitValueTyped<out T : CSSUnit>(
     override val value: Float,
     override val unit: T
 ) : CSSSizeValue<T> {
-    override fun newUnit(value: Float): CSSSizeValue<T> = copy(value = value)
-    override fun toString(): String = asString()
+    override fun toString(): String = "$value$unit"
 }
 
-operator fun <T : CSSUnit> CSSSizeValue<T>.times(num: Number): CSSSizeValue<T> = newUnit(value * num.toFloat())
-operator fun <T : CSSUnit> Number.times(unit: CSSSizeValue<T>): CSSSizeValue<T> = unit.newUnit(unit.value * toFloat())
-
-operator fun <T : CSSUnit> CSSSizeValue<T>.div(num: Number): CSSSizeValue<T> = newUnit(value / num.toFloat())
-
-operator fun <T: CSSUnit> CSSSizeValue<T>.plus(b: CSSSizeValue<T>): CSSSizeValue<T> = newUnit(value + b.value)
-operator fun <T: CSSUnit> CSSSizeValue<T>.minus(b: CSSSizeValue<T>): CSSSizeValue<T> = newUnit(value - b.value)
-
-interface CSSUnitLengthOrPercentage: CSSUnit
-interface CSSUnitPercentage: CSSUnitLengthOrPercentage
-interface CSSUnitLength: CSSUnitLengthOrPercentage
-interface CSSUnitRel : CSSUnitLength
-interface CSSUnitAbs: CSSUnitLength
-interface CSSUnitAngle: CSSUnit
-interface CSSUnitTime: CSSUnit
-interface CSSUnitFrequency: CSSUnit
-interface CSSUnitResolution: CSSUnit
-interface CSSUnitFlex: CSSUnit
+external interface CSSUnitLengthOrPercentage: CSSUnit
+external interface CSSUnitPercentage: CSSUnitLengthOrPercentage
+external interface CSSUnitLength: CSSUnitLengthOrPercentage
```

Comment:
Would "CssUnitSize" be more appropriate name?

Reply:
There already is such name. I didn't change naming here, so let's discuss naming further? I think we can improve it but it isn't about this PR.